### PR TITLE
Closes #2498 update data("sdtm") in templates and vignettes

### LINKED ISSUE #2498: Update use of data("sdtm") to sdtm <- pharmaverse::sdtm in templates and vignettes
## Documentation Update

Update use of data("sdtm") to  sdtm <- pharmaverse::sdtm

### QUESTION about lazy loading db:

In section 7.1 of R-pkgs book (https://r-pkgs.org/data.html#sec-data-data),  the use of data() is discouraged  when `lazy loading` is used (This is case for `admiral`,  `pharamversesdtm' and others, see DESCRIPTION, `LazyData: true`)

> It is important to note that lazily-loaded datasets do not need to be pre-loaded with utils::data() and, in fact, it’s usually best to avoid doing so. Above, once we did library(nycflights13), we could immediately access flights. There is no call to data(flights), because it is not necessary.

Is use of data()  still necessary?

Thx.

### Definition of Done

- [ ] All Vignettes have been updated to use sdtm <- pharmaverse::sdtm convention
- [ ] All Templates have been updated to use sdtm <- pharmaverse::sdtm convention
- [ ] Review Examples to see if any updates are needed as well if it makes sense to update (discuss once first two are completed)

### PR DESCRIPTION
**Pull request by jeffreyad**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF and a PNG version of it to the same folder. (The PNG version can be created by taking a screenshot of the PDF version.)
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by jimrothstein** [comment]
@jeffreyad 
Looks very good; much clearer the original of each dataset

File: NEWS.md
Comment (jimrothstein) [inline NEWS.md:1]: Minor point:
Would `replace` be more accurate than `update`?
```
Replace use of `data("sdtm")` ....
```

File: vignettes/hys_law.Rmd
Comment (jimrothstein) [inline vignettes/hys_law.Rmd:1]: Doing two steps at once.   It is fine with me, but differs from pattern for most of the datasets.

**Review by jimrothstein — APPROVED** [review]
Good job.

File: inst/templates/ad_adae.R [hunk 1]
```diff
@@ -14,12 +14,11 @@ library(lubridate)
...
 
-data("ae")
-data("admiral_adsl")
-data("ex_single")
-data("suppae")
+ae <- pharmaversesdtm::ae
+suppae <- pharmaversesdtm::suppae
 
-adsl <- admiral_adsl
+adsl <- admiral::admiral_adsl
```
Comment (bms63) [inline inst/templates/ad_adae.R:20]: can remove the blank space?

File: inst/templates/ad_adlb.R [hunk 2]
```diff
@@ -15,10 +15,9 @@ library(stringr)
 # as needed and assign to the variables below.
 # For illustration purposes read in admiral test data
 
-data("lb")
-data("admiral_adsl")
+lb <- pharmaversesdtm::lb
 
-adsl <- admiral_adsl
+adsl <- admiral::admiral_adsl
```
Comment (bms63) [inline inst/templates/ad_adlb.R:20]: Please remove blank space - if it is there!.

File: inst/templates/ad_adpc.R [hunk 3]
```diff
@@ -19,15 +19,12 @@ library(pharmaversesdtm) # Contains example datasets from the CDISC pilot projec
...
-data("ex")
-data("vs")
-
-data("admiral_adsl")
+pc <- pharmaversesdtm::pc
+ex <- pharmaversesdtm::ex
+vs <- pharmaversesdtm::vs
 
-adsl <- admiral_adsl
+adsl <- admiral::admiral_adsl
```
Comment (bms63) [inline inst/templates/ad_adpc.R:27]: blank space - i can't make update :(

**Review by bms63 — COMMENTED** [review]
Hey Jeff could you remove any blank spaces when loading the data.

and add a blank space between the loading section and convert_to_nas() calls?  

Just to keep everything consistent and tidy.

**Review by bms63 — APPROVED** [review]
Thanks @jeffreyad !!

**Pull request merged**